src/yea_wandb/artifact_emu.py

```diff
@@ -60,6 +61,30 @@ def create(self, variables):
 
         return response
 
+    def link(self, variables):
+        pfolio_name = variables.get("artifactPortfolioName")
+        artifact_id = variables.get("artifactID")
+        if artifact_id is None:
+            artifact_id = variables.get("clientID")
+        aliases = variables.get("aliases")
+        # TODO: assert artifact exists in our _artifacts_by_id, if not change response
+        art = {
+            "id": artifact_id,
+            "aliases": [a["alias"] for a in aliases]
+        }
+
+        # We automatically create a portfolio for the user if we can't find the one given.
+        links = self._portfolio_links.setdefault(pfolio_name, [])
+        if not any(map(lambda x: artifact_id == x["id"], links)):
+            links.append(art)
+
+        self._ctx["portfolio_links"].setdefault(pfolio_name, {})
+        self._ctx["portfolio_links"][pfolio_name]["num"] = len(links)
+        num = self._ctx["portfolio_links"][pfolio_name]["num"]
```

Comment:
nit: (just to make it easier to read)
```suggestion
        num = len(links)
        self._ctx["portfolio_links"][pfolio_name]["num"] = num
```

Reply:
nice, changed in next commit